File: services/api/src/kew_api/schemas/document.py
"""Document read/write schemas."""

from __future__ import annotations

from pydantic import BaseModel, Field


class DocumentMeta(BaseModel):
    path: str
    title: str
    checksum: str
    last_modified: str
    size_bytes: int


class DocumentResponse(BaseModel):
    path: str
    content: str
    checksum: str
    meta: DocumentMeta


class UpdateDocumentRequest(BaseModel):
    path: str = Field(min_length=1)
    content: str
    checksum: str | None = Field(
        default=None,
        description="Expected checksum for optimistic concurrency; omit to skip check",
    )


class UpdateDocumentResponse(BaseModel):
    path: str
    checksum: str
    meta: DocumentMeta
    disk_path: str | None = Field(
        default=None,
        description="Absolute filesystem path of the saved file",
    )

File: apps/workbench/src/shared/types/document.ts
export type EditorViewMode = "edit" | "preview" | "split";

export interface DocumentMeta {
  path: string;
  title: string;
  checksum: string;
  lastModified: string;
  sizeBytes: number;
}

export interface DocumentContent {
  path: string;
  content: string;
  checksum: string;
  meta: DocumentMeta;
}

export interface UpdateDocumentRequest {
  path: string;
  content: string;
  checksum?: string | null;
}

export interface UpdateDocumentResponse {
  path: string;
  checksum: string;
  meta: DocumentMeta;
}
